lib/connections/service.go

```diff
@@ -282,16 +271,11 @@ next:
 		// If rate limiting is set, and based on the address we should
 		// limit the connection, then we wrap it in a limiter.
 
-		limit := s.shouldLimit(c.RemoteAddr())
-
 		wr := io.Writer(c)
-		if limit && s.writeRateLimit != nil {
-			wr = NewWriteLimiter(c, s.writeRateLimit)
-		}
-
 		rd := io.Reader(c)
-		if limit && s.readRateLimit != nil {
-			rd = NewReadLimiter(c, s.readRateLimit)
+		if s.shouldLimit(c.RemoteAddr()) {
+			wr = s.limiter.newWriteLimiter(c)
+			rd = s.limiter.newReadLimiter(c)
 		}
```

Comment:
The classification could be done on connect, the application of classification could be done on each read write. As in, we decide if its lan or not at the point we wrap, and limit based on whats set inthe config and if the conn is classified as lan